spec/integration/type/tidy_spec.rb

```diff
@@ -11,6 +11,28 @@
     Puppet::Util::Storage.stubs(:store)
   end
 
+  it "should be able to recursively remove directories" do
+    dir = tmpfile("tidy_testing")
+    foo = File.join(dir, "foo")
+    bar = File.join(foo, "bar")
+
+    Dir.mkdir(dir)
+    Dir.mkdir(foo)
+    Dir.mkdir(bar)
+
+    tidy = Puppet::Type.type(:tidy).new :path => dir, :recurse => true, :rmdirs => true
+
+    catalog = Puppet::Resource::Catalog.new
+    catalog.add_resource(tidy)
+    # avoid crude failures because of nil resources that result
+    # from implicit containment and lacking containers
+    catalog.stubs(:container_of).returns tidy
```

Comment:
Yeah, I don't know what in here is "weird legacy stuff" and just cargo culted it. I could double-check that this line is necessary